### DISCUSSION TITLE
Node.js Custom Plugin | getUserProperties()

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#412] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12

### Description

Hello,

I would like to have more details about the getUserProperties() and the getClientType() methods of the library.
Where are they looking for this fields and what fields are they looking for?

[Documentation](https://docs.mia-platform.eu/docs/development_suite/api-console/api-design/plugin_baas_4)

**Comment by fredmaggiowski** [comment, marked as answer]
Hi @georgev-97,

- `getUserProperties()` takes the value from the header specified in the [`USER_PROPERTIES_HEADER_KEY`](https://github.com/mia-platform/custom-plugin-lib/blob/master/index.js#L58C6-L58C32)
- `getClientType()` takes the value from the header specified in the [`CLIENTTYPE_HEADER_KEY`](https://github.com/mia-platform/custom-plugin-lib/blob/master/index.js#L69)

These headers are initially set by the [Authorization Service](https://docs.mia-platform.eu/docs/runtime_suite/authorization-service/configuration) and are based on how the request is being evaluated: the _userproperties_ header is set with a json holding data extracted from the provided user access token while the _client type_ header is set according the api key client type specified in console (if a valid the `client-type` header is provided with the original request)

**Reply by georgev-97** [reply]
If I add the headers manually, will they be removed by the authorization service?

**Reply by fredmaggiowski** [reply]
Yes they will be overwritten, unless you configure the [trust Mia headers feature](https://docs.mia-platform.eu/docs/runtime_suite/authorization-service/usage#trust-mia-platform-user-headers) which can be useful in certain scenarios but introduces severe security implications, depending on your infrastructure setup (I'd not recommend using it unless you are 100% sure that requests incoming in your project are already authenticated by a third party and can be trusted as secure).

**Reply by georgev-97** [reply]
Thank you very much, that was useful!

**Comment by ajmalfaris11** [comment]
It seems like you're encountering issues with a specific product's documentation or support. Unfortunately, I'm unable to identify the exact product or library based on the information provided ("Console" and "12" could refer to various products or versions).

To address your concerns, here's a general approach you can take to get more details about the `getUserProperties()` and `getClientType()` methods:

1. **Check Official Documentation**: Start by reviewing the official documentation of the product or library you're using. Look for documentation related to these methods, including their usage, parameters, return values, and any other relevant details.

2. **Look for Code Examples**: Search for code examples or tutorials that demonstrate the usage of these methods. Sometimes, practical examples can provide more insights into how these methods are used and what they do.

3. **Review Source Code (if available)**: If the product or library is open-source, consider reviewing the source code to understand how these methods are implemented. This can provide valuable insights into their behavior and logic.

4. **Reach out to Support**: If you're unable to find satisfactory answers through documentation or code examples, consider reaching out to the product's support team or community forums for assistance. They may be able to provide clarification or additional information about these methods.

5. **Check for Updates**: Ensure that you're using the latest version of the product or library. Sometimes, documentation or behavior may change with updates, so it's important to stay up-to-date.

Without more context about the specific product or library you're using, I'm unable to provide more targeted assistance. If you can provide more details or clarify the product/library you're referring to, I'd be happy to offer more specific guidance.
